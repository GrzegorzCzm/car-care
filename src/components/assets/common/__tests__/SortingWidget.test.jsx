import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SortingWidgetWithStyles, { SortingWidget } from '../SortingWidget';


describe('SortingWidget', () => {
  const props = {
    sortParams: {
      field: 'data',
      type: 'data',
      order: 'DESC',
    },
    setSortParams: jest.fn(),
  };


  it('should renders correctly', () => {
    const propsWithClasses = {
      classes: {
        paramLine: 'expenses-paramLine',
      },
      ...props,
    };

    const wrapper = shallow(<SortingWidget {...propsWithClasses} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should contains map style in classes prop', () => {
    const expectedClasses = { paramLine: 'SortingWidget-paramLine-1' };

    const wrapper = shallow(<SortingWidgetWithStyles {...props} />);

    expect(wrapper.prop('classes')).toEqual(expectedClasses);
  });
});
