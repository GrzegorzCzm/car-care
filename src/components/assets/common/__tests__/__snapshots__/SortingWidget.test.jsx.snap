// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingWidget should renders correctly 1`] = `
<Styled(MuiBox)
  className="expenses-paramLine"
>
  <WithStyles(ForwardRef(IconButton))
    aria-label="DESC"
    onClick={[Function]}
    size="small"
  >
    <KeyboardArrowDownIcon
      color="action"
      fontSize="large"
    />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(IconButton))
    aria-label="ASC"
    onClick={[Function]}
    size="small"
  >
    <KeyboardArrowUpIcon
      color="disabled"
      fontSize="large"
    />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(FormControl))>
    <WithStyles(ForwardRef(Select))
      id="sortingFieldSelect"
      labelId="sortingFieldSelect"
      onChange={[Function]}
      value="data"
    >
      <WithStyles(ForwardRef(MenuItem))
        key="date"
        value="date"
      >
        date
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        key="item"
        value="item"
      >
        item
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        key="cost"
        value="cost"
      >
        cost
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        key="address"
        value="address"
      >
        address
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </WithStyles(ForwardRef(FormControl))>
</Styled(MuiBox)>
`;
